satpy/multiscene.py

```diff
@@ -69,8 +80,86 @@ def stack(datasets, weights=None, combine_times=True):
     return base
 
 
-def _stack_weighted(datasets, weights, combine_times):
-    """Stack datasets using weights."""
+def _stack_blended_bands(datasets, weights, combine_times):
+    """Stack datasets with bands blending overlap using weights."""
+    weights = set_weights_to_zero_where_invalid_red(datasets, weights)
+
+    dims = datasets[0].dims
+    attrs = combine_metadata(*[x.attrs for x in datasets])
+
+    if combine_times:
+        if 'start_time' in attrs and 'end_time' in attrs:
+            attrs['start_time'], attrs['end_time'] = _get_combined_start_end_times(*[x.attrs for x in datasets])
+
+    total = weights[0].copy() + 1.e-9
+    for n in range(1, len(weights)):
+        total += weights[n]
```

Comment:
```suggestion
    total = sum(weights)
```
the sum function even has a "start" argument, where you could put the offset you have here, like `total = sum(weights, start=1.e-9)`
However, I think this is not necessary. A division by zero later on will result in a NaN value, which will then be replaced with the `fillna` call.